modules/renter/repairscanner.go

```diff
@@ -306,7 +306,26 @@ func (r *Renter) threadedRepairScan() {
 	}
 	defer r.tg.Done()
 
+	// Helper function that closes a channel when we are online
+	closeWhenOnline := func(channel chan struct{}) {
+		for !r.g.Online() {
+			select {
+			case <-r.tg.StopChan():
+				break
+			case <-time.After(time.Second):
+			}
+		}
+		close(channel)
+	}
+
 	for {
+		// Check if we are online. If not, wait until we are
+		if !r.g.Online() {
+			wait := make(chan struct{})
+			go closeWhenOnline(wait)
+			<-wait
```

Comment:
why a channel here? Wouldn't this work just as well?
```go
for {
    // Wait until we are online
    for !r.g.Online() {
        select {
        case <-r.tg.StopChan():
            return
        case <-time.After(time.Second):
        }
    }

    // Refresh the worker pool and get the set of hosts that are currently
    // useful for uploading.
    hosts := r.managedRefreshHostsAndWorkers()

    // etc...
```

Reply:
oh, you still need to check `StopChan` in the main loop body though. Otherwise if `r.g.Online` returns true, you'll never check `StopChan`

Reply:
Yes, it would. I'll change that. I just thought it looks nicer not having the double for loop :P